Content.Server/VendingMachines/VendingMachineSystem.cs

```diff
@@ -384,7 +385,22 @@ private void EjectItem(EntityUid uid, VendingMachineComponent? vendComponent = n
                 return;
             }
 
-            var ent = Spawn(vendComponent.NextItemToEject, Transform(uid).Coordinates);
+            // Default spawn coordinates
+            var spawnCoordinates = Transform(uid).Coordinates;
+
+            //Make sure the wallvends spawn outside of the wall.
+            if (HasComp<WallMountComponent>(uid))
+            {
+                if (TryComp<WallMountComponent>(uid, out var wallMountComponent))
+                {
+                    const float distanceFromWallVend = 0.5f;
```

Comment:
put this constant at the top of the file with the dependencies and fix the naming convention. I'd also recommend giving it a more descriptive name and potentially increasing it to 0.75 or 1 (Since its spawning partially in the wall and getting moved)

Reply:
Had a little test 0.5/0.75/1
![image](https://github.com/space-wizards/space-station-14/assets/47093363/3d0ef0e3-7f40-4d0b-900f-d1d71f6424ab)

Needs 1f to not collide, updated, renamed and moved const up.